File: src/services/instructions/loadstore/LD_S_REGX_MREGY_XX.java
package emulator.source.loadstore

import emulator.engine.CpuContext
 import Instruction from '../Instruction'import REGISTER_VALUE_NAME_MAPPER from '@/constants/registers'

export default class LD_S_REGX_MREGY_XX extends Instruction {
	public LD_S_REGX_MREGY_XX(memory, address, source, 
 destination) {
		super(memory, address, source, destination)
		super.setArgument32()
		super.setAssembler("ld.s " + this.sdest + ", [" + this.ssource + " + 0x%08x]")
	}
	
	
	exec ({ context, memory }) {
		int address = fix(context.getReg(this.source)  + this.argument)
		context.getReg(this.destination)  = context.memory[address / 2] & 0xFFFF
		context.pc  += 6 
	}
}

File: src/services/instructions/loadstore/LdSRegXMRegYXX.js
import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination, ssource) => `ld.s ${sdestination}, [${ssource} + %s]`

export default class LdSRegXMRegYXX extends Instruction {
	constructor (memory, address, source, destination, symbolTable) {
		super(memory, address, source, destination, symbolTable)
		super.setArgument32(memory)
		super.setAssembler(ASSEMBLER_INSTRUCTION_EXPRESSION(this.sdestination, this.ssource), symbolTable)
	}

	exec ({ context, memory }) {
		const address = Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.source]]  + this.argument)
		context[REGISTER_VALUE_NAME_MAPPER[this.destination]]  = memory[address / 2] & 0xFFFF
		context.pc  += 6
	}
}
